plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoClientConfig.java

```diff
@@ -21,34 +21,34 @@
 import io.airlift.configuration.ConfigSecuritySensitive;
 import io.airlift.configuration.DefunctConfig;
 
+import javax.validation.constraints.AssertTrue;
 import javax.validation.constraints.Min;
 import javax.validation.constraints.NotNull;
-import javax.validation.constraints.Size;
 
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.mongodb.MongoCredential.createCredential;
 
-@DefunctConfig("mongodb.connection-per-host")
+@DefunctConfig({"mongodb.connection-per-host", "mongodb.socket-keep-alive", "mongodb.connection-timeout"})
```

Comment:
If I remember correctly, the old connection timeout was moved to SocketSettings's `readTimeout` in MongoDB driver. Could you take a look at their documentation?

Reply:
Yes we don't need connection timeout and socket timeout

Reply:
I'm not saying we don't need connection timeout. If "the old connection timeout was moved to SocketSettings's readTimeout in MongoDB driver", we should keep it instead of moving to defunct config.